src/warehouses/components/WarehouseSettings/WarehouseSettings.tsx

```diff
@@ -0,0 +1,188 @@
+import { WarehouseDetailsPageFormData } from "./../WarehouseDetailsPage";
+
+export interface WarehouseInfoProps {
+  zones: WarehouseDetails_warehouse_shippingZones_edges_node[];
+  disabled: boolean;
+  data: WarehouseDetailsPageFormData;
+  onShippingZoneClick: (id: string) => void;
+  onChange: (event: React.ChangeEvent<any>) => void;
+}
+
+const useStyles = makeStyles(
+  theme => ({
+    link: {
+      "&:not(:last-of-type)": {
+        marginBottom: theme.spacing()
+      }
+    }
+  }),
+  {
+    name: "WarehouseInfoProps"
+  }
+);
+
+const WarehouseInfo: React.FC<WarehouseInfoProps> = ({
+  zones,
+  disabled,
+  data,
+  onShippingZoneClick,
+  onChange
+}) => {
+  const classes = useStyles({});
+  const intl = useIntl();
+
+  const isPrivateChoices = [
+    {
+      label: (
+        <>
+          <FormattedMessage defaultMessage="Private Stock" />
+          <Typography variant="caption" color="textSecondary">
+            <FormattedMessage defaultMessage="If enabled stock in this warehouse won't be shown" />
+          </Typography>
+          <FormSpacer />
+        </>
+      ),
+      value: "true"
+    },
+    {
+      label: (
+        <>
+          <FormattedMessage defaultMessage="Public Stock" />
+          <Typography variant="caption" color="textSecondary">
+            <FormattedMessage defaultMessage="If enabled stock in this warehouse will be shown" />
+          </Typography>
+        </>
+      ),
+      value: "false"
+    }
+  ];
+
+  const clickAndCollectChoicesPublic = [
+    {
+      label: (
+        <>
+          <FormattedMessage defaultMessage="Disabled" />
+          <Typography variant="caption" color="textSecondary">
+            <FormattedMessage defaultMessage="If selected customer won't be able to choose this warehouse as pickup point" />
+          </Typography>
+          <FormSpacer />
+        </>
+      ),
+      value: WarehouseClickAndCollectOptionEnum.DISABLED
+    },
+    {
+      label: (
+        <>
+          <FormattedMessage defaultMessage="Local stock only" />
+          <Typography variant="caption" color="textSecondary">
+            <FormattedMessage defaultMessage="If selected customer will be able to choose this warehouse as pickup point. Ordered products will be only fulfilled from this warehouse stock" />
+          </Typography>
+          <FormSpacer />
+        </>
+      ),
+      value: WarehouseClickAndCollectOptionEnum.LOCAL
+    },
+    {
+      label: (
+        <>
+          <FormattedMessage defaultMessage="All warehouses" />
+          <Typography variant="caption" color="textSecondary">
+            <FormattedMessage defaultMessage="If selected customer will be able to choose this warehouse as pickup point. Ordered products can be shipped here from a different warehouse" />
+          </Typography>
+        </>
+      ),
+      value: WarehouseClickAndCollectOptionEnum.ALL
+    }
+  ];
+
+  const clickAndCollectChoicesPrivate = clickAndCollectChoicesPublic.filter(
```

Comment:
You're getting `private` collection points from a list of `public` ones, that doesn't seem to make sense. Not sure if I do not understand this correctly or the naming is incorrect.

Reply:
Why it doesnt make sense? Please note, that according to design the additional radio button `Local` should be between `ALL` and `Disabled` That is why i cannot simply put `cont clickAndCollectChoicesPublic = [...clickAndCollectChoicesPrivate, {...localChoice}]` for smth similar.

Reply:
I'm talking in general - extracting private data from public defeats the purpose of having a `private` in name (if it's available in `public`). Can you explain the difference between a private collection point and a public one?

Reply:
Hi, this is not private / public in terms of encapsulation. The possible choices depends on the field isPrivate, because in the future it will describes, whether stocks are visible for customer or not. If not (`isPrivate == true`) The local option will not have a sense, since it means that every product in `Checkout` is available from a single, given warehouse. Should I change the name of the variable, what do you think?

Reply:
I have change the name of the variable, as we spoke yesterday.